diff --git a/src/com.vixalien.decibels.in b/src/com.vixalien.decibels.in
index 3c6c521..9bf98df 100755
--- a/src/com.vixalien.decibels.in
+++ b/src/com.vixalien.decibels.in
@@ -1,6 +1,7 @@
 #!@GJS@ -m
 
 import GLib from "gi://GLib";
+import Gio from "gi://Gio";
 import { exit } from "system";
 
 imports.package.init({
@@ -10,10 +11,22 @@ imports.package.init({
   libdir: "@libdir@",
   datadir: "@datadir@",
 });
+pkg.name = "@APPLICATION_ID@";
 
 pkg.initGettext();
 pkg.initFormat();
 
+Gio.resources_register(
+    Gio.resource_load(
+        "@datadir@/@APPLICATION_ID@/com.vixalien.decibels.src.gresource",
+    ),
+);
+Gio.resources_register(
+    Gio.resource_load(
+        "@datadir@/@APPLICATION_ID@/com.vixalien.decibels.data.gresource",
+    ),
+);
+
 const loop = new GLib.MainLoop(null, false);
 import("resource://@resource_path@/js/main.js")
   .then((main) => {
